src/Form/Validation.elm

```diff
@@ -133,31 +135,37 @@ validateString schema v =
                 [ Validation.whenJust schema.minLength validateMinLength
                 , Validation.whenJust schema.maxLength validateMaxLength
                 , Validation.whenJust schema.pattern validatePattern -- TODO: check specs if this is correct
-                , Validation.whenJust schema.format validateFormat -- TODO: check specs if this is correct
+                , Validation.whenJust schema.format (validateFormat settings) -- TODO: check specs if this is correct
                 ]
                 s
 
 
-validateFormat : String -> String -> Validation String
-validateFormat format v =
+validateFormat : Settings -> String -> String -> Validation String
+validateFormat settings format value =
     case format of
         "date-time" ->
-            validateRegex Form.Regex.dateTime Error.DateTime v
+            validateRegex Form.Regex.dateTime Error.DateTime value
 
         "date" ->
-            validateRegex Form.Regex.date Error.Date v
+            validateRegex Form.Regex.date Error.Date value
 
         "time" ->
-            validateRegex Form.Regex.time Error.Time v
+            validateRegex Form.Regex.time Error.Time value
 
         "email" ->
-            validateRegex Form.Regex.email Error.Email v
+            validateRegex Form.Regex.email Error.Email value
 
         "phone" ->
-            validateRegex Form.Regex.phone Error.Phone v
+            validateRegex Form.Regex.phone Error.Phone value
 
-        _ ->
-            Validation.succeed v
+        customFormat ->
+            let
+                customValidation =
+                    Dict.get customFormat settings.customFormats
+                        |> Maybe.map (\validation -> validation value)
+                        |> Maybe.withDefault (Result.Ok value)
+            in
+            Result.mapError (\err -> error (Error.InvalidCustomFormat err)) customValidation
```

Comment:
I think it'd be good to output the format, not `err`, in `Error.InvalidCustomFormat`. Our front-end may want to show a hint according to which Personal number was requested (similar to what ID Check does):

<img width="754" height="120" alt="Image" src="https://github.com/user-attachments/assets/1a57eed2-a68c-4718-9bbe-b4cb019ac7ad" />